app/controllers/general_shopping_lists_controller.rb

```diff
@@ -0,0 +1,16 @@
+class GeneralShoppingListsController < ApplicationController
+  before_action :find_user
+
+  def index
+    @user = current_user
+    @food = @user.foods.left_outer_joins(:recipe_foods).where(recipe_foods: { id: nil }).includes(:recipe_foods)
+    @food_counts = @food.count
+    @food_prices = @food.sum(:price)
+  end
+
+  private
+
+  def find_user
+    @user = current_user
+  end
```

Comment:
- [ ] @team kindly delete this method, devise already provides us with the `current_user`  method by default, check the [documentation](https://github.com/heartcombo/devise#controller-filters-and-helpers) for more details.

Reply:
### **Thank you Mr @danmainah for your comment and for bringing this to my attention!🥰😘😘** 
We appreciate your suggestion to remove the method and use the current_user method provided by Devise instead😉😊